### PR TITLE
Test another super edge case

### PR DESCRIPTION
**Pull request by wied03**

### CODE REVIEW

File: language/super_spec.rb [hunk 1]
```diff
@@ -167,6 +167,11 @@ def a(arg)
     Super::AliasWithSuper::Trigger.foo.should == [:b, :a]
   end
 
+  it "finds super on other objects if a singleton class aliased the method" do
+    Super::SingletonAliasCase::Foo.new.alias_on_singleton
+    Super::SingletonAliasCase::Foo.new.method([]).should == [:foo, :base]
```
Comment (eregon) [inline language/super_spec.rb]: Should it call `new_method`?

File: language/fixtures/super.rb [hunk 2]
```diff
@@ -493,4 +493,25 @@ class IncludesIntermediate
 
     def foobar; 42; end
   end
+
+  module SingletonAliasCase
+    class Base
+      def method(array)
```
Comment (eregon) [inline language/fixtures/super.rb]: `method` already exists on Kernel, could you rename it?

File: language/super_spec.rb (hunk 1 above)
Comment (wied03) [inline language/super_spec.rb]: @eregon - That might be a good test as well but the case that was breaking is the original method. I did just add another test that uses a singleton'ed method though

File: language/fixtures/super.rb (hunk 2 above)
Comment (wied03) [inline language/fixtures/super.rb]: @eregon  - will rename and rebase/squash

**Comment by wied03** [comment]
@eregon - Renamed, clarified the case you mentioned, and added another case in there. Should be good to go.

**Comment by eregon** [comment]
Thank you for your contribution!
